src/components/error/idle-warning.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { useEffect, useState } from "react";
+import "./error.scss";
+
+interface IProps {
+  username: string;
+  onTimeout: () => void;
+  onContinue: () => void;
+  onExit: () => void;
+  timeout: number;
+}
+
+export const IdleWarning: React.FC<IProps> = ({ username, timeout, onTimeout, onContinue, onExit }) => {
+  const [ time, setTime ] = useState(0);
+
+  useEffect(() => {
+    const startTime = Date.now();
+    const intervalId = setInterval(() => {
+      const currentTime = Date.now();
+      const timeDiff = currentTime - startTime;
+      setTime(timeDiff);
+      if (timeDiff >= timeout) {
+        window.clearInterval(intervalId);
+      }
+    }, Math.min(timeout / 2, 500));
+    return () => window.clearInterval(intervalId);
+  }, [timeout]);
+
+  useEffect(() => {
+    if (time >= timeout) {
+      onTimeout();
+    }
+  }, [onTimeout, time, timeout]);
+  
+  const timeLeft = timeout - time;
+  const sec = Math.round((timeLeft / 1000) % 60);
+  const min = Math.floor((timeLeft / 1000) / 60);
+
+  return (
+    <div className="error" data-cy="idle-warning">
+      <h1>{username }, your session is about to expire.</h1>
+      <p>Session has been idle over its time limit.</p>
```

Comment:
Is this the wording from LARA? There's probably a more user-friendly way to say this instead of "Session has been idle over its time limit." (especially since our audience is kids).

Reply:
@pjanik I'd suggest just removing "Session has been idle over its time limit.".  That seems to be just restating the heading text with different words. 

I'd guess a more concise message means students are more likely to read all of the message. :)

Reply:
Thanks, updated.